File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.php
class Solution {

    /**
     * @param String $s
     * @return String
     */
    function longestPalindrome($s) {
        $length = strlen($s);
        
        if($length <= 1){
            return $s;
        }
        
        if(strrev($s) === $s){
            return $s;
        }
        
        $result = "";

       for($i = 0; $i < $length; $i++){
           $longest = "";
           $temp = "";

           for($j = $i; $j < $length; $j++){
               $temp .= $s[$j];

               if($temp === strrev($temp)){
                   $longest = $temp;
               }
           }

           if(strlen($longest) > strlen($result)){
               $result = $longest;
           }
       }

       return $result;
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.go
func longestPalindrome(s string) string {
    var result string
    
    for i := 0; i < len(s); i++ {
        result = palindrome(s, i, i, result)
        result = palindrome(s, i, i+1, result)
    }
    
    return result
}

func palindrome(s string, i int, j int, result string) string {
    var substring string
    
    for i >= 0 && j < len(s) && s[i] == s[j] {
        substring = s[i:j+1]
        i--
        j++
    }
    
    if len(substring) > len(result) {
        return substring
    }
    
    return result
}